test.js

```diff
@@ -217,6 +217,22 @@ describe("ECIES", function() {
     });
   });
 
+  it("should encrypt and decrypt", function() {
+    return eccrypto.encrypt(publicKeyA, Buffer.from("to a")).then(function(enc) {
+      return eccrypto.decrypt(privateKeyA, enc);
+    }).then(function(msg) {
+      expect(msg.toString()).to.equal("to a");
+    });
+  });
+
+  it("should encrypt and decrypt with message size > 15", function() {
```

Comment:
Have you tested to see if there is a size large enough that breaks it again?

Reply:
@JBaczuk I have done some tests with 30kb and it looks fine from my end. But there is still a browser bug, sometimes we get BAD Mac in the browser but it is not related to this issue.